osu.Server.Queues.ScoreStatisticsProcessor/Commands/Performance/PerformanceCommand.cs

```diff
@@ -103,7 +103,6 @@ protected async Task ProcessPartitioned<T>(IEnumerable<T> values, Func<T, Task>
             await Task.WhenAll(Partitioner
                                .Create(values)
                                .GetPartitions(Threads)
-                               .AsParallel()
```

Comment:
I am not 100% clear about *when* you're supposed to use `AsParallel` but it only seems to be required to make synchronous enumerables into asynchronous ones, which is not what we need here.

It has no effect on processing performance at all.

Reply:
I'm very confused. Why is this not what we want?

Here's a simple console app:
```csharp
using System.Collections.Concurrent;
using System.Diagnostics;

Stopwatch sw = Stopwatch.StartNew();

await processPartitioned(Enumerable.Range(0, 1000), t =>
{
    Thread.Sleep(100);
    return Task.CompletedTask;
}, CancellationToken.None);

sw.Stop();
Console.WriteLine(sw.ElapsedMilliseconds);

async Task processPartitioned<T>(IEnumerable<T> values, Func<T, Task> processFunc, CancellationToken cancellationToken)
{
    await Task.WhenAll(Partitioner
                       .Create(values)
                       .GetPartitions(8)
                       .AsParallel()
                       .Select(processPartition));

    async Task processPartition(IEnumerator<T> partition)
    {
        using (partition)
        {
            while (partition.MoveNext())
            {
                if (cancellationToken.IsCancellationRequested)
                    return;

                await processFunc(partition.Current);
            }
        }
    }
}
```
With the `AsParallel()`, this takes around 13s to complete, which is expected given the delay. Without, it does not complete for a long time.

Reply:
You're using a synchronous (non-`async`) delegate function there, which is where the difference lies. It's not required for `async` functions.

ie. try replacing the `Thread.Sleep` with an `await Task.Delay`.

Reply:
Right, sure. But isn't that what we want?

That is, regardless of if my delegate is:
```
Console.WriteLine(t);
Thread.Sleep(100);
```
or
```
Console.WriteLine(t);
await Task.Delay(100);
```

I'm getting the same throughput with `AsParallel()`, whereas not without it. Did you run into some sort of issue that made you change this or something?

Reply:
We are using `async` everywhere, so it's just not necessary. That's all.

Reply:
🙈